src/apps/rs232_test/app_main.cpp

```diff
@@ -0,0 +1,315 @@
+                  // once done initializing everything
+                  2, NULL);
+  configASSERT(rval == pdTRUE);
+
+  // Start FreeRTOS scheduler
+  vTaskStartScheduler();
+
+  /* We should never get here as control is now taken by the scheduler */
+
+  while (1) {
+  };
+}
+
+bool start_stress(const void *pinHandle, uint8_t value, void *args) {
+  (void)pinHandle;
+  (void)args;
+  static bool started;
+
+  if (!started && value) {
+    started = true;
+    stress_start_tx((2 * 1024) / 8);
+  }
+
+  return false;
+}
+
+void handle_subscriptions(uint64_t node_id, const char *topic,
+                          uint16_t topic_len, const uint8_t *data,
+                          uint16_t data_len, uint8_t type, uint8_t version) {
+  (void)node_id;
+  if (strncmp(APP_PUB_SUB_UTC_TOPIC, topic, topic_len) == 0) {
+    if (type == APP_PUB_SUB_UTC_TYPE && version == APP_PUB_SUB_UTC_VERSION) {
+      utcDateTime_t time;
+      const bm_common_pub_sub_utc_t *utc =
+          reinterpret_cast<const bm_common_pub_sub_utc_t *>(data);
+      dateTimeFromUtc(utc->utc_us, &time);
+
+      RTCTimeAndDate_t rtc_time = {
+          .year = time.year,
+          .month = time.month,
+          .day = time.day,
+          .hour = time.hour,
+          .minute = time.min,
+          .second = time.sec,
+          .ms = (time.usec / 1000),
+      };
+
+      if (rtcSet(&rtc_time) == pdPASS) {
+        printf("Set RTC to %04u-%02u-%02uT%02u:%02u:%02u.%03u\n", rtc_time.year,
+               rtc_time.month, rtc_time.day, rtc_time.hour, rtc_time.minute,
+               rtc_time.second, rtc_time.ms);
+      } else {
+        printf("\n Failed to set RTC.\n");
+      }
+    } else {
+      printf("Unrecognized version: %u and type: %u\n", version, type);
+    }
+  } else {
+    printf("Topic: %.*s\n", topic_len, topic);
+    printf("Data: %.*s\n", data_len, data);
+  }
+}
+
+// TODO - move this to some debug file?
+// Defines list if GPIOs for Debug CLI
+static const DebugGpio_t debugGpioPins[] = {
+    {"adin_cs", &ADIN_CS, GPIO_OUT},
+    {"adin_int", &ADIN_INT, GPIO_IN},
+    {"adin_pwr", &ADIN_PWR, GPIO_OUT},
+    {"adin_rst", &ADIN_RST, GPIO_OUT},
+    {"pl_buck_en", &PL_BUCK_EN, GPIO_OUT},
+    {"vbus_en", &VBUS_EN, GPIO_OUT}, // Should I remove the bb_ and just call it vbus_en?
+    {"flash_cs", &FLASH_CS, GPIO_OUT},
+    {"boot_led", &BOOT_LED, GPIO_IN},
+    {"vusb_detect", &VUSB_DETECT, GPIO_IN},
+    {"discharge_on", &DISCHARGE_ON, GPIO_OUT},
+};
+
+static void defaultTask(void *parameters) {
+  (void)parameters;
+
+  mxInit();
+
+  usbMspInit();
+
+  rtcInit();
+
+  // Initialize low power manager
+  lpmInit();
+
+  // Inhibit low power mode during boot process
+  lpmPeripheralActive(LPM_BOOT);
+
+  startIWDGTask();
+  startSerial();
+
+  startSerialConsole(&usbCLI);
+  // Serial device will be enabled automatically when console connects
+  // so no explicit serialEnable is required
+  pcapInit(&usbPcap);
+
+  startCLI();
+  startDebugUart();
```

Comment:
This app is a blend of bringup + bristlemouth features. It has all of the bristlemouth features, but does not use the "user_code" and "payload_uart" to collect data from a sensor. Instead it has debug_uart included and you need to use `serial enable lpuart` serial baud lpuart 9600` to be able to talk to an RBR. 

However, the power monitor running on the I2C bus is running automatically with the power/sensor sampler. 

If this feels a bit redundant, then perhaps I can just create a "serial_payload_example" app and get rid of this app. 

Or if we need a true bringup app, then I can create that here.

Reply:
I talked with Sophie and she menitoned that as far as production goes the mote goes through bringup and then after the expander board is attached there really isn't another step of bringup requiring the ability to turn things like I2C/serial on/off.

So I think just creating a serial_payload_example app will suffice. These apps already have the ability to toggle the gpio's on/off so I think that will be enough for most testing.

Reply:
With that in mind I'm thinking to not merge this app as is, but convert it into a "bm_rs232_apps/serial_payload_example" type of app. Thoughts?

Reply:
In the latest commits I have created a serial_payload_example to run on the rs232 and removed the rs232_test app.

Reply:
I can't revert to the commit where both the serial_paylaod_example and rs232_test app exist if we want both.

Reply:
No need to revert — that test app was just my WIP, making sure I understood everything. I like serial_payload_example as the final thing that stays in the code. 👍